pyzeebe/task/task_builder.py

```diff
@@ -1,80 +1,78 @@
-import inspect
+import functools
 import logging
-from typing import List, Callable, Dict, Tuple
+from typing import Awaitable, Callable, Dict, Sequence, Tuple
 
-from pyzeebe import Job, TaskDecorator
+from pyzeebe import Job
+from pyzeebe.function_tools.async_tools import asyncify, is_async_function
+from pyzeebe.function_tools.dict_tools import convert_to_dict_function
 from pyzeebe.task.task import Task
 from pyzeebe.task.task_config import TaskConfig
-from pyzeebe.task.types import DecoratorRunner, JobHandler
+from pyzeebe.task.types import AsyncTaskDecorator, DecoratorRunner, JobHandler
 from pyzeebe.worker.task_state import TaskState
 
 logger = logging.getLogger(__name__)
 
 
 def build_task(task_function: Callable, task_config: TaskConfig) -> Task:
-    if task_config.single_value:
-        task_function = convert_to_dict_function(
-            task_function, task_config.variable_name)
-
     return Task(task_function, build_job_handler(task_function, task_config), task_config)
 
 
 def build_job_handler(task_function: Callable, task_config: TaskConfig) -> JobHandler:
+    prepared_task_function = prepare_task_function(task_function, task_config)
+
     before_decorator_runner = create_decorator_runner(task_config.before)
     after_decorator_runner = create_decorator_runner(task_config.after)
 
-    def job_handler(job: Job, task_state: TaskState = None) -> Job:
+    @functools.wraps(task_function)
+    async def job_handler(job: Job, task_state: TaskState = None) -> Job:
         if task_state:
             task_state.add(job)
```

Comment:
This happens in [pyzeebe/worker/job_poller.py](https://github.com/camunda-community-hub/pyzeebe/pull/172/files#diff-6fe1ff288c0d1911f34a6c397bd05247b28e9cf083ba6b29ac22af4fc0b78516R41)